src/link-manager.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Array that stores the preset view names of all the views.
+ */
+/** @type {!Array<string>} */
+genotet.linkManager.presetStatus = [];
```

Comment:
Ok, I will create another object to store such (action, response) pairs. However, this object should be renamed to presetViews to store the preset views and their type. Link Manager should know which views are preset or not.